internal/ethapi/api.go

```diff
@@ -1098,17 +1099,22 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		result.L1MessageSender = meta.L1MessageSender
 		if meta.QueueOrigin != nil {
 			switch meta.QueueOrigin.Uint64() {
-			case uint64(2):
-				result.QueueOrigin = "sequencer"
+			case uint64(types.QueueOriginSequencer):
```

Comment:
The type should be decoupled from the `QueueOrigin` and deserialized from the tx data when it is unknown.